### PR TITLE
Implement checkout update quantity product and fix comment review

### PR DESCRIPTION
**Pull request by thanhtrantruong47**
Implement checkout update quantity product and fix comment review

- Handle checkout when api error
- Delete banner in component MainLayout
- Fix bug selected quantity max value

### CODE REVIEW

File: src/hook/CartContext.tsx [hunk 1]
```diff
@@ -1,18 +1,24 @@
...
 import { CartItem as CartItemType } from '../types/cartItem';
 
 interface CartContextType {
   productsInCart: CartItemType[];
   addToCart: (item: CartItemType) => void;
   updateQuantity: (id: string, newQuantity: number) => void;
   removeFromCart: (id: string) => void;
-  clearCart: () => void; // Add clearCart to the context type
+  clearCart: () => void;
+  isRemoving: string | null; // Track which item is being removed
```
Comment (hungdo-agilityio) [inline src/hook/CartContext.tsx]: can we use optional here `isRemoving?: string`

File: src/components/CartItem/CartItem.tsx [hunk 2]
```diff
@@ -5,6 +5,7 @@ import DeleteIcon from '../Icon/DeleteIcon';
 import Button from '../Button/Button';
 import { useState } from 'react';
 import unAvailableImage from '../../assets/images/no-image.jpg';
+import Loading from '../Loading';
```
Comment (hungdo-agilityio) [inline src/components/CartItem/CartItem.tsx]:
```
import Button from '../Button/Button'
import Loading from '../Loading'
```
Please consistent import, you can fix in another PR

File: src/components/CartItem/CartItem.tsx [hunk 3]
```diff
@@ -30,47 +31,61 @@ const CartItem = ({
...
     onQuantityChange(id, newQuantity);
   };
 
   const handleClick = () => {
+    setIsLoading(true);
     onChangeDelete(id);
+
+    setTimeout(() => {
+      setIsLoading(false);
+    }, 500);
```
Comment (hungdo-agilityio) [inline src/components/CartItem/CartItem.tsx:47]: Why we need to setTimeout here

File: src/hook/CartContext.tsx (hunk 1 above)
Comment (thanhtrantruong47) [reply src/hook/CartContext.tsx]: Fixed

File: src/components/CartItem/CartItem.tsx (hunk 3 above)
Comment (thanhtrantruong47) [reply src/components/CartItem/CartItem.tsx:47]: I need it to simulate the api call process

Comment (hungdo-agilityio) [reply src/components/CartItem/CartItem.tsx:47]: Please note comments if you mock or do something cause confused

File: src/components/CartItem/CartItem.tsx (hunk 2 above)
Comment (thanhtrantruong47) [reply src/components/CartItem/CartItem.tsx]: Fixed

File: src/components/CartItem/CartItem.tsx (hunk 3 above)
Comment (thanhtrantruong47) [reply src/components/CartItem/CartItem.tsx:47]: fixed: add comment code

File: src/components/FooterNavList/FooterNavList.tsx [hunk 4]
```diff
@@ -14,9 +14,15 @@ const FooterNavList = ({ listItemFooter, title }: FooterNavListProps) => {
       <ul className={styles.item}>
         {listItemFooter.map(({ label, destination }) => (
           <li key={label}>
-            <Link className={styles.link} to={destination}>
-              {label}
-            </Link>
+            {destination.startsWith('mailto:') ? (
```
Comment (thanhnguyen-agilityio) [inline src/components/FooterNavList/FooterNavList.tsx]: recheck

File: src/components/NumberInput/NumberInput.tsx [hunk 5]
```diff
@@ -8,11 +8,14 @@ import Button from '../Button/Button';
 interface NumberInputProps {
   value: number; // Current quantity
   onChange: (value: number) => void; // Callback to handle quantity changes
+  max: number; // Maximum value
 }
 
-const NumberInput = ({ value, onChange }: NumberInputProps) => {
+const NumberInput = ({ value, onChange, max }: NumberInputProps) => {
```
Comment (thanhnguyen-agilityio) [inline src/components/NumberInput/NumberInput.tsx]: `max` -> `maxValue`

**Review by thanhnguyen-agilityio — APPROVED** [review]

**Pull request merged**